api-v2/src/student/student.controller.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  ParseUUIDPipe,
+  BadRequestException,
+  UseGuards,
+  Req,
+  Query,
+  ParseBoolPipe,
+  DefaultValuePipe,
+} from '@nestjs/common';
+import { StudentService } from './student.service';
+import { CreateStudentDto } from './dto/create-student.dto';
+import { UpdateStudentDto } from './dto/update-student.dto';
+import { DeleteResult, UpdateResult } from 'typeorm';
+import { Student } from './entities/student.entity';
+import { JwtAuthGuard } from 'src/guards/jwt-auth.guard';
+import { DevRouteGuard } from 'src/guards/dev-route.guard';
+import { AuthenticatedRequest } from 'src/auth/interfaces/authenticated-request';
+
+@Controller('students')
+export class StudentController {
+  constructor(private readonly studentService: StudentService) {}
+
+  @UseGuards(JwtAuthGuard)
+  @Get('/me')
+  async getMe(
+    @Req() req: AuthenticatedRequest,
+    @Query('isWithPlans', new DefaultValuePipe(false), ParseBoolPipe)
```

Comment:
what does this do?

Reply:
Haha not sure if this is about the "isWithPlan" search param or the default value pipe or the parse bool pipe so imma just explain all 3.
- isWithPlan is a boolean search param that tells us whether we need to return a student with their plans or not. Implemented this so that we can don't always perform table joins when providing students unless we need. Not sure if this is useful yet(maybe for pages where we just display username or something?) but I just default to providing an option like this when dealing with multiple tables. This could be argued so feel free to if you have thoughts!
- DefaultValuePipe(false) makes the param `false` by default if it isn't specified
- ParseBoolPipe extracts the string for the search param and makes sure it is "true" or "false". If responds with an error saying `isWithPlans` should be "true" or "false". If it is a valid string then it parses it into a boolean depending on the string.

Reply:
Did any of the above 3 things answer the question haha?

Reply:
yep!